examples/mnist_tfrecord.py

```diff
@@ -0,0 +1,115 @@
+'''MNIST dataset with TensorFlow TFRecords.
+
+Gets to 99.25% test accuracy after 12 epochs
+(there is still a lot of margin for parameter tuning).
+'''
+import os
+import copy
+import time
+
+import numpy as np
+
+import tensorflow as tf
+from keras import backend as K
+from keras.models import Model
+from keras.layers import Dense
+from keras.layers import Dropout
+from keras.layers import Flatten
+from keras.layers import Input
+from keras.layers import Conv2D
+from keras.layers import MaxPooling2D
+from keras.objectives import categorical_crossentropy
+from keras.utils import np_utils
+from keras.utils.generic_utils import Progbar
+from keras import callbacks as cbks
+from keras import optimizers, objectives
+from keras import metrics as metrics_module
+
+from tensorflow.contrib.learn.python.learn.datasets.mnist import load_mnist
+
+if K.backend() != 'tensorflow':
+    raise RuntimeError('This example can only run with the '
+                       'TensorFlow backend for the time being, '
+                       'because it requires TFRecords, which '
+                       'are not supported on other platforms.')
+
+
+def cnn_layers(x_train_input):
+    x = Conv2D(32, (3, 3), activation='relu', padding='valid')(x_train_input)
+    x = Conv2D(64, (3, 3), activation='relu')(x)
+    x = MaxPooling2D(pool_size=(2, 2))(x)
+    x = Dropout(0.25)(x)
+    x = Flatten()(x)
+    x = Dense(128, activation='relu')(x)
+    x = Dropout(0.5)(x)
+    x_train_out = Dense(classes,
+                        activation='softmax',
+                        name='x_train_out')(x)
+    return x_train_out
+
+
+sess = tf.Session()
+K.set_session(sess)
+
+batch_size = 256
+batch_shape = [batch_size, 28, 28, 1]
+steps_per_epoch = 1000
+epochs = 3
+classes = 10
+capacity = 10000
+min_after_dequeue = 3000
+
+data = load_mnist()
+x_train_batch, y_train_batch = tf.train.shuffle_batch(
```

Comment:
@TimZaman Based on a meeting I had with @fchollet I'm just making this into a minimal PR that conforms with existing well known methods that should work out of the box with Keras and TensorFlow. This just incorporates some essential functionality and examples in a minimal way, since #6928 is a change set that is several times larger.